src/main/java/uk/gov/ida/matchingserviceadapter/resources/MatchingServiceResource.java

```diff
@@ -41,7 +77,41 @@ public MatchingServiceResource(
     public Response receiveSoapRequest(Document attributeQueryDocument) {
         LOG.debug("AttributeQuery POSTED: {}", attributeQueryDocument);
 
-        MatchingServiceResponse matchingServiceResponse = matchingService.handle(new MatchingServiceRequestContext(attributeQueryDocument));
-        return responseGenerator.generateResponse(matchingServiceResponse);
+        AttributeQuery attributeQuery = unwrapAttributeQuery(attributeQueryDocument);
+        List<Assertion> assertions = assertionDecrypter.decryptAssertions(new EncryptedAssertionContainer(attributeQuery));
+
+        // If no assertions are present, assume this is a Health Check
+        if (assertions.isEmpty()) {
+            return responseGenerator.generateHealthCheckResponse(attributeQuery.getID());
```

Comment:
Do we want people to be able to send healthchecks that don't have valid signatures? This looks like it comes before the signature validation.